cypress/integration/focus-trap-demo.spec.js

```diff
@@ -381,22 +423,98 @@ describe('focus-trap', () => {
+      verifyCrucialFocusTrapOnClicking('@radioB');
+
+      /*
+       radio group value can be changed
+      */
+
+      // Cypress limitation to radio group value change: keyboard arrow keys can't change value
+      // so forcibly change the value by `check` command
+      //    cy.findByRole('group', { name: 'Radio group in trap' }).type('{downArrow}');
+      //    cy.get('@radioB').type('{downArrow}');
+      cy.get('@testRoot')
+        .findByRole('radio', { name: 'c' })
+        .as('radioC')
+        .check()
+        .should('be.checked');
+
+      // 'Tab' in trap should only focus the checked radio item without changing radio group value
+      cy.get('@radioC').focus();
+      cy.tab();
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'deactivate trap' })
+        .as('deactivateElInTrap')
+        .should('be.focused');
+      cy.tab();
+      cy.get('@radioC').should('be.focused');
+
+      // focus can be transitioned freely when trap is deactivated
+      cy.get('@deactivateElInTrap').click();
+      verifyFocusIsNotTrapped(cy.get('@lastlyFocusedElBeforeTrapIsActivated'));
+    });
   });
 
   describe('demo: iframe', () => {
-    // TODO
+    it('focus can be moved freely in the iframe inside of trap', () => {
+      cy.get('#demo-iframe').as('testRoot');
+
+      // activate trap and element trap should be focused
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'activate trap' })
+        .as('lastlyFocusedElBeforeTrapIsActivated')
+        .click();
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'nothing' })
+        .as('firstElInTrap')
+        .should('be.focused');
+      verifyCrucialFocusTrapOnClicking('@firstElInTrap');
+
+      /*
+       focus can move into/out of iframe in trap
+       Due to cypress limitation(see below), `tab` is not tested
+      */
+
+      // Cypress limitation: `cy.tab()` skips focus on <iframe>
+      // cy.tab();
+      cy.get('@testRoot').find('iframe').as('iframeInTrap');
+      cy.get('@iframeInTrap').then(([iframe]) => {
+        // Cypress limitation: TypeError: Cannot read property 'call' of undefined
+        // cy.wrap(iframe.contentDocument).find('a').should('exist');
+        const buttonEl = iframe.contentDocument.querySelector('button');
+        cy.wrap(buttonEl).as('buttonInIFrame').click();
+        cy.get('@buttonInIFrame').should('be.focused');
+
+        verifyCrucialFocusTrapOnClicking('@buttonInIFrame');
+      });
+
+      // focus can be transitioned freely when trap is deactivated
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'deactivate trap' })
+        .as('deactivateElInTrap')
+        .click();
+      verifyFocusIsNotTrapped(cy.get('@lastlyFocusedElBeforeTrapIsActivated'));
+    });
   });
 
   describe('demo: allowoutsideclick', () => {
-    // TODO
+    // TODO: defect? this only traps keyboard but click on outside element is still changing document.activeElement
```

Comment:
This is the only left test case I am not sure about. The keyboard `tab` is trapped when it's activated but click on the demo page is not trapped at all. Will come back to this one in the next few days.

Reply:
This trap allows clicking outside of it while maintaining an active trap, if I remember correctly. That would explain why our "verify focus is trapped" utility function doesn't work, because the utility function expects that clicking on those repo links will do nothing -- but they end-up trying to load the page, and I'll bet you were seeing the test fail with an error from Cypress talking about some cross-domain error thingamagig.

I ran into the same thing when I recently added another demo and test case [clickoutsidedeactivates](https://github.com/focus-trap/focus-trap/blob/master/cypress/integration/focus-trap-demo.spec.js#L402) which is closely related to the pre-existing "allowoutsideclick" demo. I wrote a test-specific [checkTrap()](https://github.com/focus-trap/focus-trap/blob/master/cypress/integration/focus-trap-demo.spec.js#L410) helper because, for this test, since it also allows outside clicks, we can only verify that focus is trapped by running the expected tab sequence back and forth...

Reply:
Ooh okay now I get the idea. I will add one test for this.